client/landing/stepper/declarative-flow/internals/analytics/record-design.ts

```diff
@@ -0,0 +1,90 @@
+import { SiteIntent } from '@automattic/data-stores/src/onboard';
+import { Design, StyleVariation } from '@automattic/design-picker/src';
+import { resolveDeviceTypeByViewPort } from '@automattic/viewport';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+
+export function recordPreviewedDesign( {
+	flow,
+	intent,
+	design,
+	styleVariation,
+}: {
+	flow: string | null;
+	intent: string;
+	design: Design;
+	styleVariation?: StyleVariation;
+} ) {
+	recordTracksEvent( 'calypso_signup_design_preview_select', {
+		...getDesignEventProps( { flow, intent, design, styleVariation } ),
+		goes_to_assembler_step: design.design_type === 'assembler',
```

Comment:
[nit] Just assume that maybe `design_type: 'assembler'` is enough 😄

Reply:
@autumnfjeld wants this prop 😂